### PR TITLE
Always default version labels so we can retire the 1.0.0 value

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/18349

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-go 1.15
+go 1.16
 
 require (
 	github.com/Masterminds/semver/v3 v3.1.1
 	github.com/dyson/certman v0.2.1
 	github.com/giantswarm/apiextensions/v3 v3.30.0
-	github.com/giantswarm/app/v4 v4.13.0
+	github.com/giantswarm/app/v5 v5.0.2-0.20210806135610-a9a53e24985b
```
Comment (rossf7) [inline go.mod:9]: Needs https://github.com/giantswarm/app/pull/136

**Comment by rossf7** [comment]
Tested on gauss.

File: pkg/app/mutate_app.go [hunk 2]
```diff
@@ -124,26 +122,34 @@ func (m *Mutator) MutateApp(ctx context.Context, oldApp, app v1alpha1.App, opera
...
-	labelPatches, err := m.mutateLabels(ctx, app, appVersionLabel)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-	if len(labelPatches) > 0 {
-		result = append(result, labelPatches...)
+		if patchLabels {
+			m.logger.Debugf(ctx, "mutating only labels of app %#q in namespace %#q due to version label %#q", app.Name, app.Namespace, appVersionLabel)
+			return result, nil
+		} else {
```
Comment (tomahawk28) [inline pkg/app/mutate_app.go:149]: I think you don't need this `else` statement.

File: pkg/app/mutate_app_test.go [hunk 3]
```diff
@@ -260,6 +260,37 @@ func Test_MutateApp(t *testing.T) {
...
+					Name:      "kiam",
+					Namespace: "kube-system",
+					KubeConfig: v1alpha1.AppSpecKubeConfig{
+						InCluster: true,
+					},
+					Version: "1.4.0",
+				},
+			},
+			apps: []*v1alpha1.App{
+				newTestApp("chart-operator", "eggs2", "3.1.0"),
```
Comment (tomahawk28) [inline pkg/app/mutate_app_test.go:286]: Can you add a test for which we don't have `chart-operator` app CR yet?

File: pkg/app/mutate_app.go (hunk 2 above)
Comment (rossf7) [reply pkg/app/mutate_app.go:149]: I think I was still in holiday mode :) I'll fix.

File: pkg/app/mutate_app_test.go (hunk 3 above)
Comment (rossf7) [reply pkg/app/mutate_app_test.go:286]: Good catch. I'll add.

File: go.mod [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
-go 1.15
+go 1.16
 
 require (
 	github.com/Masterminds/semver/v3 v3.1.1
 	github.com/dyson/certman v0.2.1
 	github.com/giantswarm/apiextensions/v3 v3.32.0
-	github.com/giantswarm/app/v4 v4.13.0
-	github.com/giantswarm/app/v5 v5.1.0
+	github.com/giantswarm/app/v5 v5.2.1-0.20210816140705-8ff37814ffc3
```
Comment (rossf7) [inline go.mod:9]: Needs https://github.com/giantswarm/app/pull/146

**Review by rossf7 — COMMENTED** [review]
Tested on gauss

@tomahawk28 PTAL

**Review by tomahawk28 — APPROVED** [review]
LGTM!

**Pull request merged**